geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/IdExtractingVisitor.scala

```diff
@@ -42,13 +42,10 @@ class IdExtractingVisitor extends FilterVisitor {
     (andOption(yes).orNull, andOption(no).orNull)
   }
 
-  override def visit(f: Not, data: AnyRef): AnyRef = {
-    val (yes, no) = f.getFilter.accept(this, data).asInstanceOf[(Filter, Filter)]
-    (if (yes == null) null else ff.not(yes), if (no == null) null else ff.not(no))
```

Comment:
before it was recursing into NOTs and extracting any IDs from there - now if we see a NOT we just stop and return no match. It's possible we might miss some double-negative, but detecting that would be kind of complicated I think.